src/lib/es6.d.ts

```diff
@@ -127,7 +127,7 @@ interface ObjectConstructor {
       * @param target The target object to copy to.
       * @param source The source object from which to copy properties.
       */
-    assign<T, U>(target: T, source: U): T & U;
+    assign<T, U>(target: T, source: U | null | undefined): T & U;

```

Comment:
do not think this is the correct change here. the current signature declaration is better. i think what you want is a type operator like the expression operator `!` (which does not exist) on the return type to indicate that null and undefined are ignored. e.g. `assign<T, U>(target: T, source: U): T! & U!;`